src/qibo/quantum_info/clifford.py

```diff
@@ -0,0 +1,328 @@
+from dataclasses import dataclass
+from functools import reduce
+from itertools import product
+
+import numpy as np
+
+from qibo import Circuit
+from qibo.backends import CliffordBackend
+from qibo.config import raise_error
+from qibo.gates import M
+from qibo.measurements import frequencies_to_binary
+
+
+def _string_product(operators):
+    """Calculates the tensor product of a list of operators represented as strings.
+
+    Args:
+        operators (list): The list of operators.
+
+    Returns:
+        product (str): The string representing the tensor product of the operators.
+    """
+    # calculate global sign
+    phases = np.array(["-" in op for op in operators], dtype=bool)
+    # remove the - signs
+    operators = "|".join(operators).replace("-", "").split("|")
+    prod = []
+    for op in zip(*operators):
+        tmp = "".join([o for o in op if o != "I"])
+        if tmp == "":
+            tmp = "I"
+        prod.append(tmp)
+    result = "-" if len(phases.nonzero()[0]) % 2 == 1 else ""
+    return f"{result}({')('.join(prod)})"
+
+
+def _list_of_matrices_product(operators):
+    """Calculates the product of a list of operators as np.ndarrays.
+
+    Args:
+        operators (list): The list of operators.
+
+    Returns:
+        product (np.ndarray): The tensor product of the operators.
+    """
+    return reduce(np.matmul, operators)  # faster
+    # return np.einsum(*[d for i, op in enumerate(operators) for d in (op, (i, i + 1))])
+
+
+@dataclass
+class Clifford:
```

Comment:
`Clifford()` should initiate the object with the symplectic matrix of the zero state